apps/dashboard/src/components/side-navigation/side-navigation.tsx

```diff
@@ -128,49 +128,81 @@ export const SideNavigation = () => {
-                <RiBarChartBoxLine className="size-4" />
-                <span>Activity Feed</span>
-              </NavigationLink>
-            </NavigationGroup>
-            <NavigationGroup label="Developer">
-              <NavigationLink to={buildRoute(ROUTES.API_KEYS, { environmentSlug: currentEnvironment?.slug ?? '' })}>
-                <RiKey2Line className="size-4" />
-                <span>API Keys</span>
-              </NavigationLink>
-              {isWebhooksManagementEnabled && (
-                <NavigationLink to={buildRoute(ROUTES.WEBHOOKS, { environmentSlug: currentEnvironment?.slug ?? '' })}>
-                  <RiSignalTowerLine className="size-4" />
-                  <span className="flex items-center gap-2">Webhooks</span>
+              </Protect>
+              <Protect permission={PermissionsEnum.SUBSCRIBER_READ}>
+                <NavigationLink
+                  to={buildRoute(ROUTES.SUBSCRIBERS, { environmentSlug: currentEnvironment?.slug ?? '' })}
+                >
+                  <RiGroup2Line className="size-4" />
+                  <span>Subscribers</span>
                 </NavigationLink>
+              </Protect>
+              {isTopicsPageActive && (
+                <Protect permission={PermissionsEnum.TOPIC_READ}>
+                  <NavigationLink to={buildRoute(ROUTES.TOPICS, { environmentSlug: currentEnvironment?.slug ?? '' })}>
+                    <RiDiscussLine className="size-4" />
+                    <span>Topics</span>
+                  </NavigationLink>
+                </Protect>
               )}
-              <NavigationLink to={buildRoute(ROUTES.ENVIRONMENTS, { environmentSlug: currentEnvironment?.slug ?? '' })}>
-                <RiDatabase2Line className="size-4" />
-                <span>Environments</span>
-              </NavigationLink>
-              <NavigationLink to={buildRoute(ROUTES.INTEGRATIONS, { environmentSlug: currentEnvironment?.slug ?? '' })}>
-                <RiStore3Line className="size-4" />
-                <span>Integration Store</span>
-              </NavigationLink>
             </NavigationGroup>
+            <Protect permission={PermissionsEnum.NOTIFICATION_READ}>
+              <NavigationGroup label="Monitor">
+                <Protect permission={PermissionsEnum.NOTIFICATION_READ}>
+                  <NavigationLink
+                    to={buildRoute(ROUTES.ACTIVITY_FEED, { environmentSlug: currentEnvironment?.slug ?? '' })}
+                  >
+                    <RiBarChartBoxLine className="size-4" />
+                    <span>Activity Feed</span>
+                  </NavigationLink>
+                </Protect>
+              </NavigationGroup>
+            </Protect>
+            <Protect
+              condition={(has) =>
+                has({ permission: PermissionsEnum.API_KEY_READ }) ||
+                has({ permission: PermissionsEnum.INTEGRATION_READ }) ||
+                has({ permission: PermissionsEnum.WEBHOOK_READ })
```

Comment:
Just as a general note, I wonder if it will be cleaner to allow an array on "has" ?

Reply:
The original Clerk `<Protect>` component which we are just wrapping/extending has it like that, so I wanted to preserve the original and not introduce new abstraction on top of that for now.

https://clerk.com/docs/components/protect#render-content-by-permissions

Reply:
That makes sense, thanks